### PR TITLE
Update docs regarding creating work queues and agent start commands

### PR DESCRIPTION
**Pull request by jlowin**
Updating documentation that references creating work queues (no longer necessary!) and starting agents to use the latest conventions.

Reflects changes made in #6406

### CODE REVIEW

File: docs/concepts/work-queues.md [hunk 1]
```diff
@@ -12,40 +12,103 @@ tags:
     - concurrency limits
 ---
 
-# Work Queues and Agents
+# Agents and Work Queues
```
Comment (tpdorsey) [inline docs/concepts/work-queues.md:15]: Looks like the title is set in mkdocs.yaml and needs to be changed there so "Agents and Work Queues" appears in the nav

File: docs/concepts/work-queues.md [hunk 2]
```diff
@@ -12,40 +12,103 @@ tags:
...
+    It's helpful to think of work queues as a way to coordinate (potentially many) deployments with (potentially many) agents through a known channel: the queue itself. This is similar to how "topics" are used to connect producers and consumers in a pub/sub or message-based system. By switching a deployment's work queue, users can quickly change the agent that will execute their runs, making it easy to promote runs through environments or even debug locally.
 
-Work queues contain scheduled runs from any deployments that match the queue criteria. Criteria is based on deployment _tags_ &mdash; all runs for deployments that have the tags defined on the queue will be picked up.
+In addition, users can control aspects of work queue behavior like how many runs the queue allows to be run concurrently or pausing delivery entirely. These options can be modified at any time, and any agent processes requesting work for a specific queue will only see matching flow runs.
 
-These criteria can be modified at any time, and agent processes requesting work for a specific queue will only see matching flow runs.
+!!! warning "Work queue affinity improved starting from 2.0.5"
```
Comment (tpdorsey) [inline docs/concepts/work-queues.md:103]:
```suggestion
!!! warning "Work queue affinity improved starting from Prefect 2.0.5"
```

File: docs/concepts/work-queues.md [hunk 3]
```diff
@@ -12,40 +12,103 @@ tags:
...
+    Until Prefect 2.0.4, tags were used to associate flow runs with work queues. Given the flexibility of tags, it wasn't always clear which runs would be picked up by which work queues. Starting in 2.0.5, tag-based work queues are deprecated. Instead, work queue names are used to explicitly direct flow runs from deployments into queues.
```
Comment (tpdorsey) [inline docs/concepts/work-queues.md:104]:
```suggestion
    Until Prefect 2.0.4, tags were used to associate flow runs with work queues. Given the flexibility of tags, it wasn't always clear which runs would be picked up by which work queues. Starting in Prefect 2.0.5, tag-based work queues are deprecated. Instead, work queue names are used to explicitly direct flow runs from deployments into queues. 
```

File: docs/ui/cloud-getting-started.md [hunk 4]
```diff
@@ -237,18 +236,14 @@ To demonstrate that your deployment exists, go back to Prefect Cloud and select
 
 ### Create a work queue and agent
 
-Next, create a work queue that can distribute your new deployment to agents for execution, and a local agent to pick up the flow run for your 'Testing/Test Deployment' deployment. 
+Next, we start an agent that can pick up the flow run from your 'Testing/Test Deployment' deployment. Remember that when we created the deployment, it was configured to send work to a work queue called `test`. Work queues This work queue was automatically created when we created the deployment. In Prefect Cloud, you can view your work queues and create new ones manually by selecting the **Work Queues** page.
```
Comment (tpdorsey) [inline docs/ui/cloud-getting-started.md:239]:
```suggestion
Next, we start an agent that can pick up the flow run from your 'Testing/Test Deployment' deployment. Remember that when we created the deployment, it was configured to send work to a work queue called `test`. This work queue was automatically created when we created the deployment. In Prefect Cloud, you can view your work queues and create new ones manually by selecting the **Work Queues** page.
```

**Review by tpdorsey — APPROVED** [review]
Nice! There may be some merge weirdness with the docker and k8s tutorials. I had to rename them -- users were finding the old, outdated tutorials still through search.

File: docs/concepts/work-queues.md (hunk 1 above)
Comment (jlowin) [reply docs/concepts/work-queues.md:15]: Thanks!

**Pull request merged**